File: Module1/CaseStudy/6HuongDoiTuong/Customer.js
let Customer = function () {

    this.setNameCustomer = function (nameCustomer) {
        this.nameCustomer = nameCustomer;
    };
    this.getNameCustomer = function () {
        return this.nameCustomer;
    };

    this.setIdCard = function (idCard) {
        this.idCard = idCard;
    };
    this.getIdCard = function () {
        return this.idCard;
    };

    this.setBirthdayCustomer = function (birthdayCustomer) {
        this.birthdayCustomer = birthdayCustomer;
    };
    this.getBirthdayCustomer = function () {
        return this.birthdayCustomer;
    };

    this.setEmailCustomer = function (emailCustomer) {
        this.emailCustomer = emailCustomer;
    };
    this.getEmailCustomer = function () {
        return this.emailCustomer;
    };

    this.setAddressCustomer = function (addressCustomer) {
        this.addressCustomer = addressCustomer;
    };
    this.getAddressCustomer = function () {
        return this.addressCustomer;
    };

    this.setTypeCustomer = function (typeCustomer) {
        this.typeCustomer = typeCustomer;
    };
    this.getTypeCustomer = function () {
        return this.typeCustomer;
    };
    this.setDiscount = function (discount) {
        this.discount = discount;
    };
    this.getDiscount = function () {
        return this.discount;
    };

    this.setNumberOfAccompanying = function (numberOfAccompanying) {
        this.numberOfAccompanying = numberOfAccompanying;
    };
    this.getNumberOfAccompanying = function () {
        return this.numberOfAccompanying;
    };

    this.setTypeRoom = function (typeRoom) {
        this.typeRoom = typeRoom;
    };
    this.getTypeRoom = function () {
        return this.typeRoom;
    };

    this.setRentDays = function (rentDays) {
        this.rentDays = rentDays;
    };
    this.getRentDays = function () {
        return this.rentDays;
    };

    this.setTypeService = function (typeService) {
        this.typeService = typeService;
    };
    this.getTypeService = function () {
        return this.typeService;
    };

    // this.accompaniedService = function () {
    //     let accompaniedService = 0;
    //
    // };


    this.totalPays = function () {
        let moneyPerDays = 0;
        if (this.getTypeService().trim().toLowerCase() === "villa") {
            moneyPerDays = 500;
        } else if (this.getTypeService().trim().toLowerCase() === "house") {
            moneyPerDays = 300;
        } else if (this.getTypeService().trim().toLowerCase() === "room") {
            moneyPerDays = 100;
        }

        let result = 0;

        result = moneyPerDays * parseFloat(this.getRentDays()) * (1 - parseFloat(this.getDiscount()) / 100);

        if (this.getAddressCustomer().toLowerCase().replace(/ /g,"")  === "danang") {
            result -= 20;
        } else if (this.getAddressCustomer().toLowerCase().replace(/ /g,"")  === "hue") {
            result -= 10;
        } else if (this.getAddressCustomer().toLowerCase().replace(/ /g,"")  === "quangnam") {
            result -= 5;
        }

        if (parseInt(this.getRentDays()) >= 7) {
            result -= 30;
        } else if (parseInt(this.getRentDays()) >= 5) {
            result -= 20;
        } else if (parseInt(this.getRentDays()) >= 2) {
            result -= 10;
        }

        if (this.getTypeCustomer().trim().toLowerCase() === "diamond") {
            result -= 15;
        } else if (this.getTypeCustomer().trim().toLowerCase() === "platinum") {
            result -= 10;
        } else if (this.getTypeCustomer().trim().toLowerCase() === "gold") {
            result -= 5;
        } else if (this.getTypeCustomer().trim().toLowerCase() === "silver") {
            result -= 2;
        }

        let currentYear = new Date().getFullYear();
        let birthday = new Date(this.getBirthdayCustomer()).getFullYear();
        let ageCustomer = currentYear - birthday;

        if (ageCustomer >= 30 && this.getAddressCustomer().toLowerCase().replace(/ /,'') === "danang") {
            result -= 2;
        } else if (ageCustomer >= 20 && this.getAddressCustomer().toLowerCase().replace(/ /,"") === "danang") {
            result -= 1;
        }

        return result;


    }


};

File: Module2Core/src/CaseStudy/Models/Customer.java
package CaseStudy.Models;

public class Customer {
    private String nameCustomer;
    private String birthdayCustomer;
    private String genderCustomer;
    private int idCustomer;
    private int phoneNumberCustomer;
    private String emailCustomer;
    private String typeCustomer;
    private String addressCustomer;
    private Service service;



    public String showInfo() {
        return "Customer{" +
                "nameCustomer='" + nameCustomer + '\'' +
                ", birthdayCustomer='" + birthdayCustomer + '\'' +
                ", genderCustomer='" + genderCustomer + '\'' +
                ", idCustomer=" + idCustomer +
                ", phoneNumberCustomer=" + phoneNumberCustomer +
                ", emailCustomer='" + emailCustomer + '\'' +
                ", typeCustomer='" + typeCustomer + '\'' +
                ", typeAddressCustomer='" + addressCustomer + '\'' +
                ", service=" + service +
                '}';
    }



    public Customer() {
    }

    public Customer(String nameCustomer, String birthdayCustomer, String genderCustomer, int idCustomer, int phoneNumberCustomer, String emailCustomer, String typeCustomer, String addressCustomer, Service service) {
        this.nameCustomer = nameCustomer;
        this.birthdayCustomer = birthdayCustomer;
        this.genderCustomer = genderCustomer;
        this.idCustomer = idCustomer;
        this.phoneNumberCustomer = phoneNumberCustomer;
        this.emailCustomer = emailCustomer;
        this.typeCustomer = typeCustomer;
        this.addressCustomer = addressCustomer;
        this.service = service;
    }

    public String getNameCustomer() {
        return nameCustomer;
    }

    public void setNameCustomer(String nameCustomer) {
        this.nameCustomer = nameCustomer;
    }

    public String getBirthdayCustomer() {
        return birthdayCustomer;
    }

    public void setBirthdayCustomer(String birthdayCustomer) {
        this.birthdayCustomer = birthdayCustomer;
    }

    public String getGenderCustomer() {
        return genderCustomer;
    }

    public void setGenderCustomer(String genderCustomer) {
        this.genderCustomer = genderCustomer;
    }

    public int getIdCustomer() {
        return idCustomer;
    }

    public void setIdCustomer(int idCustomer) {
        this.idCustomer = idCustomer;
    }

    public int getPhoneNumberCustomer() {
        return phoneNumberCustomer;
    }

    public void setPhoneNumberCustomer(int phoneNumberCustomer) {
        this.phoneNumberCustomer = phoneNumberCustomer;
    }

    public String getEmailCustomer() {
        return emailCustomer;
    }

    public void setEmailCustomer(String emailCustomer) {
        this.emailCustomer = emailCustomer;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public String getAddressCustomer() {
        return addressCustomer;
    }

    public void setAddressCustomer(String addressCustomer) {
        this.addressCustomer = addressCustomer;
    }

    public Service getService() {
        return service;
    }

    public void setService(Service service) {
        this.service = service;
    }

}
